src/sentry/scripts/digests/digests.lua

```diff
@@ -297,34 +373,79 @@ local function parse_arguments(arguments)
         return string.format('%s:t:%s:r:%s', self.namespace, timeline_id, record_id)
     end
 
-    return configuration, table.slice(arguments, 4)
-end
+    return configuration
+end)
 
 local commands = {
-    SCHEDULE = function (arguments)
-        local configuration, arguments = parse_arguments(arguments)
-        return schedule(configuration, unpack(arguments))
+    SCHEDULE = function (cursor, arguments)
+        local cursor, configuration, deadline = multiple_argument_parser(
+            configuration_argument_parser,
+            argument_parser(tonumber)
+        )(cursor, arguments)
+        return schedule(configuration, deadline)
     end,
-    MAINTENANCE = function (arguments)
-        local configuration, arguments = parse_arguments(arguments)
-        return maintenance(configuration, unpack(arguments))
+    MAINTENANCE = function (cursor, arguments)
+        local cursor, configuration, deadline = multiple_argument_parser(
+            configuration_argument_parser,
+            argument_parser(tonumber)
+        )(cursor, arguments)
+        return maintenance(configuration, deadline)
     end,
-    ADD = function (arguments)
-        local configuration, arguments = parse_arguments(arguments)
-        return add_record_to_timeline(configuration, unpack(arguments))
+    ADD = function (cursor, arguments)
+        local cursor, configuration, arguments = multiple_argument_parser(
+            configuration_argument_parser,
+            object_argument_parser({
```

Comment:
This doesn't need to actually parse into an object but since this is a complicated signature I found it clearer to declare it this way, rather than a multiple assignment of 10 items at once.